### PR TITLE
Create Waiting for Quiz to Start Page (Host View)

### PR DESCRIPTION
**Pull request by yesh0907**
**[UPDATE: PLEASE LOOK AT THIS VIDEO OF ALL THE CHANGES](https://www.loom.com/share/a221531445a74a18a5f7298c764b320c?sid=ef457390-3a70-4142-ab05-fb5bfb166e5d)**

[check out this video of the changes I made](https://www.loom.com/share/97e54790af1a4e21b72ebe8720f73c4e)

using this as my Acceptance Criteria:
Screen Shot 2024-02-03 at 8 33 14 PM

### CODE REVIEW

**Comment by yesh0907** [comment]
DON'T MERGE TIL BE IS READY

**Comment by justinqbui** [comment]
I think we should use https://legendapp.com/open-source/state/intro/introduction/ for state management. Imo it is much easier to use for state management than react out of the box, and will probably make certain front end tasks a bit easier down the road.

Open to pushback on this as I'm not really a front-end person, but this was much easier for me to use when I was forced to do front-end.

**Review by justinqbui — APPROVED** [review]

File: src/routes/player/JoinRoom.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function JoinRoom() {
+    // Extract room_id from the URL
+    const { room_id: roomId } = useParams();
+
+    useEffect(() => {
+        // Check if room_id is available before emitting the message
+        if (roomId) {
+            // Emit a "joinRoom" event to the server with the roomId
+            socket.emit("joinRoom", roomId);
```
Comment (justinqbui) [inline src/routes/player/JoinRoom.tsx]: [NIT] we should move all socket events to a shared WS_Event like we have on the backend

File: src/socket/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { io } from "socket.io-client";
+
+// get WebSocket URL from env file
+const websocketUrl = import.meta.env.VITE_WEBSOCKET_URL;
```
Comment (justinqbui) [inline src/socket/index.ts]: should post what the websocketURL should look like in the discord

**Comment by yesh0907** [comment]
Legend state looks good, i am down to use it. found this sick video covering it: https://www.youtube.com/watch?v=E4TH77SMOG8&ab_channel=JackHerrington

**Comment by yesh0907** [comment]
**Summary of the changes that have been made to this PR since last reviewed:**
1. JoinRoom component has been modified to emit `WS_EVENTS.JOIN_ROOM` and `WS_EVENTS.NEW_PLAYER` upon mounting
2. WaitingForQuiz component accepts the `playerId` in a restructured JS object as per the changes I made in [this PR](https://github.com/yesh0907/Toohak-BE/pull/9)
3. You can now as a host create a room, get other players to join the room, and start the quiz [(check out this loom for a demo)](https://www.loom.com/share/a221531445a74a18a5f7298c764b320c?sid=ef457390-3a70-4142-ab05-fb5bfb166e5d)

**Pull request merged**